app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -115,11 +115,18 @@ def save_asset_detail_inventory(parent, hash)
   end
 
   #
-  # Saves the drive information of a storage
+  # Saves the disks information of a storage
   #
-  def save_physical_disk_inventory(parent, hash)
-    return if hash.nil?
-    save_inventory_single(:physical_disk, parent, hash)
+  def save_physical_disks_inventory(physical_storage, hashes, target = nil)
+    return if hashes.nil?
+    deletes = target == physical_storage ? :use_association : []
```

Comment:
For these child collections you should just use the association as long as the target type cannot an instance of this collection.
Unless you expect to actually get physical_storage as a top-level refresh target just pass `:use_association` into `save_inventory_multi`

Reply:
The end result of this will be that _no_ physical_disks will ever be deleted, because the target will always be the EMS.

Reply:
> For these child collections you should just use the association as long as the target type cannot an instance of this collection.
> Unless you expect to actually get physical_storage as a top-level refresh target just pass `:use_association` into `save_inventory_multi`

@agrare I changed it to use `:use_association` unless `PhysicalStorage` is assigned as the target since these disks are related to physical storage units, this also solves the deletion problem. Though I'm still thinking if it's interesting to use `EMS` instead of `PhysicalStorage` as default in case other providers decide to use this model. What do you think?